central/alert/datastore/datastore_impl.go

```diff
@@ -386,36 +399,53 @@ func applyDefaultState(q *v1.Query) *v1.Query {
 			storage.ViolationState_ACTIVE.String(),
 			storage.ViolationState_ATTEMPTED.String()).ProtoQuery())
 		cq.Pagination = q.GetPagination()
+		cq.Selects = q.GetSelects()
 		return cq
 	}
 	return q
 }
 
-// convertAlert returns proto search result from an alert object and the internal search result
-func convertAlert(alert *storage.ListAlert, result searchCommon.Result) *v1.SearchResult {
-	entityInfo := alert.GetCommonEntityInfo()
-	var entityName string
-	switch entity := alert.GetEntity().(type) {
-	case *storage.ListAlert_Resource:
-		entityName = entity.Resource.GetName()
-	case *storage.ListAlert_Deployment:
-		entityName = entity.Deployment.GetName()
-	}
-	resourceTypeTitleCase := strings.Title(strings.ToLower(entityInfo.GetResourceType().String()))
+// AlertSearchResultConverter implements search.SearchResultConverter for alert search results.
+// This enables single-pass query construction for SearchResult protos.
+type AlertSearchResultConverter struct{}
+
+func (c *AlertSearchResultConverter) BuildName(result *search.Result) string {
+	return result.Name
+}
+
+func (c *AlertSearchResultConverter) BuildLocation(result *search.Result) string {
+	fv := result.FieldValues
+	clusterName := fv[strings.ToLower(search.Cluster.String())]
+	namespace := fv[strings.ToLower(search.Namespace.String())]
+	deploymentName := fv[strings.ToLower(search.DeploymentName.String())]
+	resourceName := fv[strings.ToLower(search.ResourceName.String())]
+
+	var entityName, resourceType string
+	if deploymentName != "" {
+		entityName = deploymentName
+		resourceType = "Deployment"
+	} else if resourceName != "" {
+		entityName = resourceName
+		resourceType = "Resource"
+	}
+
 	var location string
-	if entityInfo.GetNamespace() != "" {
-		location = fmt.Sprintf("/%s/%s/%s/%s",
-			entityInfo.GetClusterName(), entityInfo.GetNamespace(), resourceTypeTitleCase, entityName)
-	} else {
-		location = fmt.Sprintf("/%s/%s/%s",
-			entityInfo.GetClusterName(), resourceTypeTitleCase, entityName)
-	}
-	return &v1.SearchResult{
-		Category:       v1.SearchCategory_ALERTS,
-		Id:             alert.GetId(),
-		Name:           alert.GetPolicy().GetName(),
-		FieldToMatches: searchCommon.GetProtoMatchesMap(result.Matches),
-		Score:          result.Score,
-		Location:       location,
+	if entityName != "" {
+		if namespace != "" {
+			location = fmt.Sprintf("/%s/%s/%s/%s",
+				clusterName, namespace, resourceType, entityName)
```

Comment:
This should be the resource type title case.  It shoudl be the resource type itself, not `Resource` right?

```    
enum ResourceType {
      UNKNOWN = 0;
      SECRETS = 1;
      CONFIGMAPS = 2;
      CLUSTER_ROLES = 3;
      CLUSTER_ROLE_BINDINGS = 4;
      NETWORK_POLICIES = 5;
      SECURITY_CONTEXT_CONSTRAINTS = 6;
      EGRESS_FIREWALLS = 7;
    }
```

Reply:
But why?   What is the error when using ResourceType?
and the complaint here. is that the location is either set to the word`Deployment` or the word `Resource` if it is not a deployment.  That does not seem to match the functionality in the old code.